ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ActiveContracts.scala

```diff
@@ -178,11 +193,19 @@ class ActiveContractsManager[ACS](
             }
         }
 
-    st.result
+    st.mapAcs(_ implicitlyDisclose globalImplicitDisclosure).result
   }
 
 }
 
+trait ActiveContractsSteps[+Self] { this: ActiveContractsSteps[Self] =>
+  def lookupContract(cid: AbsoluteContractId): Option[ActiveContract]
+  def keyExists(key: GlobalKey): Boolean
+  def addContract(cid: AbsoluteContractId, c: ActiveContract, keyO: Option[GlobalKey]): Self
+  def removeContract(cid: AbsoluteContractId, keyO: Option[GlobalKey]): Self
+  def implicitlyDisclose(global: Relation[AbsoluteContractId, Ref.Party]): Self
+}
+
 object ActiveContracts {
 
   case class ActiveContract(
```

Comment:
please rename `witnesses` to `disclosedTo`, as discussed.

Reply:
fb861d97506edd7a1a809749b106a7c9dbd4c381